File: readLog.js
#!/usr/local/bin/node
var async = require('async');
var fileName='../data/tempController.log';
var resetFlag=false;
var text='';
var tx=new Array();
var lineNo=0;
var i = 0;
var j = 0;
var nxpstep=9;
var nxp=0;
var dir='';

//console.log('process.argv.length :'+process.argv.length);
//console.log('resetFlag :'+resetFlag+' fileName:'+fileName+' nxp:'+nxp+' nxpstep:'+nxpstep);
// readLog.js [-f] logfile [-s] start [-reset] [-l] length

if(process.argv.length >= 3 ){
	//console.log('process.argv[2][0]:|'+process.argv[2][0]+'|');
	if(process.argv[2][0] !== '-')
		fileName=process.argv[2];
}
if(process.argv.length === 4){
	if(process.argv[2][0] !== '-')
		fileName=process.argv[2];
	if(process.argv[3][0] !== '-')
		nxp=parseInt(process.argv[3]);
}
if(process.argv.length === 5){
	if(process.argv[2][0] !== '-')
		fileName=process.argv[2];
	if(process.argv[3][0] !== '-')
		nxp=parseInt(process.argv[3]);
	if(process.argv[4][0] !== '-')
		nxpstep=parseInt(process.argv[4]);
}
process.argv.forEach(function(val, index, array) {
	if(val==='-r') resetFlag=true;
	if(val==='-s') {
			resetFlag=true; 
			nxp=parseInt(array[index+1]);
			//console.log('resetFlag:'+resetFlag);
			};
	if(val==='-f') fileName=array[index+1];
	if(val==='-l') nxpstep=parseInt(array[index+1]);
});

if(fileName.indexOf('/data/')<0 ) fileName = '../data/'+fileName;

var fs = require('fs');

var sline=false;
//console.log('resetFlag :'+resetFlag+' fileName:'+fileName+' nxp:'+nxp+' nxpstep:'+nxpstep);
//console.log('nxp :'+nxp+' nxpstep:'+nxpstep);

var fsc = require('fs-sync');

//console.log('resetFlag :'+resetFlag+' fileName:'+fileName+' nxp:'+nxp+' nxpstep:'+nxpstep);
//console.log('nxp :'+nxp+' nxpstep:'+nxpstep);

    var execSync = require('child_process').execSync;
        result =  execSync('/usr/bin/wc -l '+fileName);
        tmp=result.toString().split(' ')[0];
    var filemax=parseInt(tmp)-1;
	nxpstep=filemax;
    
    readline = require('readline'),
    rs = fs.ReadStream(fileName),
    rl = readline.createInterface({'input': rs, 'output': {}});

rl.on('line', function (line) {
	var t=line.trim();
	tx.push(t);

// parse
	if((tx.length-1)>=(filemax)){
	for(j=nxp;j<nxpstep;j++){
	//console.log('tx['+j+']: '+tx[j]);

	if( tx[j].indexOf('-------')>=0){
	sline=true;
	//console.log('sline: '+sline);
	//console.log('date: '+tx[j+1]);
	date=tx[j+1];
	j++;	
	continue;
	}

	//console.log('sline: '+sline);
	if( sline === true ){
		if( tx[j].indexOf('currentTemp')>=0){
		//console.log('temp: '+tx[j]);
		continue;
		}
		if( tx[j].indexOf('currentLux')>=0){
		//console.log('currentLux: '+tx[j]);
		c = (tx[j].split('currentLux=')[1]);
		if(!isFinite(c)) c = 0;
		continue;
		}
		if( tx[j].indexOf('PWM')>=0){
		//console.log('PWM: '+tx[j]);
		p = (tx[j].split(' '));
		if( tx[j].indexOf('hot')>=0){
		//console.log('hot: '+tx[j]);
		dir = (tx[j].split(' '));
		continue;
		}
		if( tx[j].indexOf('cool')>=0){
		//console.log('cool: '+tx[j]);
		dir = (tx[j].split(' '));
		continue;
		}
		}

		if( tx[j].indexOf('BME280')>=0){
		//console.log('BME280: '+tx[j]);
		d = (tx[j].split(' '));
		continue;
		}
		if( tx[j].indexOf('freezedFan')>=0){
		//console.log('freezedFan: '+tx[j]);
		g = (tx[j].split(' '));
		continue;
		}
		if( tx[j].indexOf('Mratio')>=0){
		//console.log('Mratio: '+tx[j]);
		t = (tx[j].split(' '));
		sline=false;
	// output
    	dr=0.0
	if(dir){
    	if(dir[1].indexOf('hot')){
        	dr = 1.0;}
    	else {
        	dr = -1.0;
		}
		
    console.log( date+','+d[5]+','+d[3]+','+d[7]+','+ (dr*parseFloat(p[1])/10.0)+','+g[1]+','+t[5]+','+t[3]+','+c);
		}// output end

		}
	}

	}	
	nxp += nxpstep;
	}

	i++;
});

rl.resume();

rs.on('end', function() {
  console.log('EOF');
  process.exit();
});


setInterval(function() {
//  console.log('WAIT');
},10);

File: readLog.py
#!/usr/bin/env python

import sys
import re
import numpy as np

start=0

filename ='tempController.log'
#print (len(sys.argv))
if(len(sys.argv)==2):
	filename=sys.argv[1]
# filename -s start_line
for i in range(1,len(sys.argv)):
	if(sys.argv[i]=='-s'):
		start=sys.argv[i+1]
#print start 

if(filename[0] != '/'):
	f = open('/home/coder/coder-dist/coder-base/data/'+filename)
else:
	f = open(filename)
#f = open('log.log')
text = f.read() 
f.close()
text = text.replace('=',' ')
lines1 = text.split('\n')

#for line in lines1:
#    print line
#print

#print(re.search('currentTemp=',lines1))

date=np.array([])
temp=np.array([])
pwm=np.array([])
dir=np.array([])
bme280=np.array([])
lux=np.array([])
cons=np.array([])
ds18b20=np.array([])
tmp=np.array([])

startcol=False
datain=False

#print len(lines1)

#for line in lines1:
#print 'start='+str(start) 

if int(start) > int(len(lines1)):
	start=len(lines1)

start=int(start)
#print 'start='+str(start) 
#print 'len='+str(len(lines1)) 

for i in range(start*9,len(lines1)):
    line=lines1[i];

    tmp=np.append(tmp,line[:])

    if line.find('--------') >= 0:
	startcol=True
	datein=False

    if line.find('20') >= 0 and startcol==True and datein==False:
	date=np.append(date,line[:])
	datein=True

    if line.find('currentTemp') >= 0:
	temp=np.append(temp,line[:])

    if line.find('PWM') >= 0:
	pwm=np.append(pwm,line[:])

    if line.find('hot') >= 0:
	dir=np.append(dir,line[:])

    if line.find('cool') >= 0:
	dir=np.append(dir,line[:])

    if line.find('BME280') >= 0:
	bme280=np.append(bme280,line[:])

    if line.find('currentLux') >= 0:
	lux=np.append(lux,line[:])

    if line.find('DS18B20') >= 0:
	ds18b20=np.append(ds18b20,line[:])

    if line.find('Tcondensation') >= 0:
	cons=np.append(cons,line[:])
	startcol=False

#for i in range(start/9,len(temp)):
#	print temp[i]

for i in range(start,len(temp)-1):
#    print date[i+1]+' '+temp[i]+' '+pwm[i]+' '+bme280[i]+' '+lux[i]+' '+cons[i]+' '+ds18b20[i]
    #print bme2801[i].split(' ')
    d = bme280[i].split(' ')
    p = pwm[i].split(' ')
    dr=0.0
    if dir[i].split(' ')[2]=='hot':
	dr = 1.0
    else:
	dr = -1.0
    #print dir1[i].split(' ')[2]+' '+'dr '+str(dr)

    g = ds18b20[i].split(' ')
    t = cons[i].split(' ')
    # date  temp hum press pwm dir constration AH 
    print date[i+1]+','+d[5]+','+d[3]+','+d[7]+','+ str(dr*float(p[1])/10.0)+','+g[1]+','+t[5]+','+t[3]

print 'EOF'
